### ISSUE TITLE
[ACID]latexmath2: problems with positioning

### ISSUE THREAD

**Issue by PeterPablo** [matlab2tikz/matlab2tikz#450]
Testcase `latexmath2` has issues with node positioning.

**Comment by okomarov** [comment]
I think this test case is out of scope. Why would I want to to typeset something on a plain figure and then port it to latex with `m2t`, when I can simply typeset it in latex directly?

![capture](https://cloud.githubusercontent.com/assets/3731173/5648837/64565800-9689-11e4-8b6d-312a4345724a.PNG)

I suggest to delete this test.

**Comment by PeterPablo** [comment]
I also vote for deletion. When I created the bug report, I was merely gathering everything going wrong in an obvious way (without judging).

**Comment by nschloe** [comment]
This test checks node placement and proper "MATLAB LaTeX" to LaTeX conversion. While I agree that you wouldn't want to do this, I don't see why we shouldn't support this situation.

**Comment by okomarov** [comment]
I dug a bit into the problem and fixed the missing units conversion for 3D text position. Unfortunately, to properly handle the 3D case, we should re-work how we determine if the axis is 3D or not. Our current way of handling is not ideal. I'll be looking into that.

**Comment by nschloe** [comment]
> Our current way of handling is not ideal.

Indeed. As far as I can recall we carry around a flag in m2t that specifies whether there is 3D data or not.

**Comment by okomarov** [comment]
After #518 I get:

![latexmath2](https://cloud.githubusercontent.com/assets/3731173/5693333/f2ef4210-990f-11e4-8122-a83462ed66ed.PNG)

**Comment by okomarov** [comment]
The `FontSize` is not picked.
